### PR TITLE
code-server: 4.0.1 -> 4.8.3

### PR DESCRIPTION
**Pull request by henkery**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/code-server/build-vscode-nogit.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
+--- ./ci/build/build-vscode.sh
++++ ./ci/build/build-vscode.sh
+@@ -45,14 +45,12 @@
+   # Set the commit Code will embed into the product.json.  We need to do this
+   # since Code tries to get the commit from the `.git` directory which will fail
+   # as it is a submodule.
+-  export VSCODE_DISTRO_COMMIT
+-  VSCODE_DISTRO_COMMIT=$(git rev-parse HEAD)
++  export VSCODE_DISTRO_COMMIT=ca182b9fb51e2b1683d6e154ba5086fc7e8c3238
```
Comment (SuperSandro2000) [inline pkgs/servers/code-server/build-vscode-nogit.patch:9]: We cannot hardcode this hear. Ideally we could just use the version attribute and do not use the hash at all.

File: pkgs/servers/code-server/default.nix [hunk 2]
```diff
@@ -21,28 +21,29 @@ let
 
 in stdenv.mkDerivation rec {
   pname = "code-server";
-  version = "4.0.1";
-  commit = "7fe23daf009e5234eaa54a1ea5ff26df384c47ac";
+  version = "4.8.1";
+  commit = "ca182b9fb51e2b1683d6e154ba5086fc7e8c3238";
```
Comment (SuperSandro2000) [inline pkgs/servers/code-server/default.nix:25]:
```suggestion
```
Purely informational

File: pkgs/servers/code-server/default.nix [hunk 3]
```diff
@@ -21,28 +21,29 @@ let
...
 
   src = fetchFromGitHub {
     owner = "cdr";
     repo = "code-server";
     rev = "v${version}";
-    sha256 = "1s3dcmzlkyh7qfs3ai1p7dlp45iys0ax1fbxxz17p395pw9anrrl";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/servers/code-server/default.nix:31]: Do we really need to do this? submodules are not known to be super reproducible

File: pkgs/servers/code-server/default.nix [hunk 4]
```diff
@@ -66,18 +67,22 @@ in stdenv.mkDerivation rec {
...
-    outputHash = "0qmfsirld1qfl2s26rxbpmvxsyj2pvzkgk8w89zlrgbhgc5fj8p9";
+    outputHash = "0jzzbmmgv1nfq975mi9ii9l6c4f1wy10fyy117xgm4s6vxana7qn";
   };
 
   nativeBuildInputs = [
-    nodejs yarn' python pkg-config makeWrapper git rsync jq moreutils
+    nodejs yarn' python pkg-config makeWrapper git rsync jq moreutils quilt
   ];
-  buildInputs = lib.optionals (!stdenv.isDarwin) [ libsecret ]
+    buildInputs = lib.optionals (!stdenv.isDarwin) [ libsecret ]
```
Comment (SuperSandro2000) [inline pkgs/servers/code-server/default.nix:85]: Please revert this change

File: pkgs/servers/code-server/default.nix [hunk 5]
```diff
@@ -133,79 +140,68 @@ in stdenv.mkDerivation rec {
...
     jq "del(.scripts) | .gypfile = false" ./package.json | sponge ./package.json
     popd
   '' + ''
-    # rebuild binaries, we use npm here, as yarn does not provide an alternative
-    # that would not attempt to try to reinstall everything and break our
-    # patching attempts
-    npm rebuild --prefix vendor/modules/code-oss-dev --update-binary
+
+    # put ripgrep binary into bin, so postinstall does not try to download it
+    find -name ripgrep -type d \
```
Comment (SuperSandro2000) [inline pkgs/servers/code-server/default.nix:199]: Can we use relative paths instead of using find?

**Review by SuperSandro2000 — COMMENTED** [review]
The commit message for the package should be ``code-server: 4.0.1 -> 4.8.1``

File: pkgs/servers/code-server/build-vscode-nogit.patch (hunk 1 above)
Comment (henkery) [reply pkgs/servers/code-server/build-vscode-nogit.patch]: replaced with dummy value instead as a value is required but it doesn't seem to matter what is put in

File: pkgs/servers/code-server/default.nix (hunk 2 above)
Comment (henkery) [reply pkgs/servers/code-server/default.nix]: removed statement

File: pkgs/servers/code-server/default.nix (hunk 3 above)
Comment (henkery) [reply pkgs/servers/code-server/default.nix:30]: upstream has changed to use submodules since previous derivation
build process requires that the submodule is pulled specifically in the directory specified so I believe it is required
if there is a way to pull this code via nix and then put it in the required location that is worth investigating but would increase maintenance complexity

File: pkgs/servers/code-server/default.nix (hunk 4 above)
Comment (henkery) [reply pkgs/servers/code-server/default.nix]: fixed

File: pkgs/servers/code-server/default.nix (hunk 5 above)
Comment (henkery) [reply pkgs/servers/code-server/default.nix:198]: this statement could be replaced but would increase maintenance complexity as the paths changed include hashes

File: pkgs/servers/code-server/default.nix [hunk 6]
```diff
@@ -21,28 +21,28 @@ let
 
 in stdenv.mkDerivation rec {
   pname = "code-server";
-  version = "4.0.1";
-  commit = "7fe23daf009e5234eaa54a1ea5ff26df384c47ac";
+  version = "4.8.1";
 
   src = fetchFromGitHub {
     owner = "cdr";
```
Comment (SuperSandro2000) [inline pkgs/servers/code-server/default.nix:27]:
```suggestion
    owner = "coder";
```
just noticed they renamed. Please update all other fetchers and links, too.

File: pkgs/servers/code-server/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/servers/code-server/default.nix:30]: using vscode as a submodule from upstream is alright.

File: pkgs/servers/code-server/default.nix [hunk 7]
```diff
@@ -21,28 +21,28 @@ let
...
+    sha256 = "0si6fzafk5skn58b83f2500za8zmqasxyma5adxhpfnxk26fg0h5";
   };
 
   cloudAgent = buildGoModule rec {
     pname = "cloud-agent";
-    version = "0.2.3";
+    version = "0.2.6";
 
     src = fetchFromGitHub {
       owner = "cdr";
```
Comment (SuperSandro2000) [inline pkgs/servers/code-server/default.nix:39]:
```suggestion
      owner = "coder";
```

File: pkgs/servers/code-server/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/servers/code-server/default.nix:198]: We could cd with wildcards into the directory. Not sure if thats better or not.

File: pkgs/servers/code-server/default.nix (hunk 7 above)
Comment (henkery) [reply pkgs/servers/code-server/default.nix]: resolved

File: pkgs/servers/code-server/default.nix (hunk 6 above)
Comment (henkery) [reply pkgs/servers/code-server/default.nix]: resolved

**Comment by Mic92** [comment]
Thanks!
